nginx_proxy/Dockerfile

```diff
@@ -1,7 +1,96 @@
 ARG BUILD_FROM
-ARG BUILD_ARCH
 
-FROM jhampdbre/${BUILD_ARCH}-nginx-116-modsecurity as build_modsecurity
+FROM ${BUILD_FROM} as build_modsecurity
+
+ARG NGINX_VERSION=1.16.1
+ARG MODSEC_BRANCH=v3/master
+ARG OWASP_BRANCH=v3.3/master
+
+WORKDIR /opt
+
+# Install dependencies; includes dependencies required for compile-time options:
+# curl, libxml, pcre, and lmdb and Modsec
+# https://build.alpinelinux.org/buildlogs/build-3-11-x86_64/main/nginx/nginx-1.16.1-r6.log
+RUN echo "Installing Dependencies" && \
+    apk add --no-cache --virtual general-dependencies \
+    alpine-sdk \
+    autoconf \
+    automake \
+    byacc \
+    curl-dev \
+    findutils \
+    flex \
+    g++ \
+    gcc \
+    gd-dev \
+    geoip-dev \
+    git \
+    libc-dev \
+    libedit-dev \
+    libstdc++ \
+    libtool \
+    libxml2-dev \
+    libxslt-dev \
+    linux-headers \
+    lmdb-dev \
+    make \
+    libmaxminddb-dev \
+    openssl-dev \
+    pcre-dev \
+    perl-dev \
+    yajl-dev \
+    zlib-dev
+
+# Clone and compile modsecurity. Binary will be located in /usr/local/modsecurity
+RUN echo "Installing ModSec Library" && \
+    git clone -b ${MODSEC_BRANCH} --depth 1 https://github.com/SpiderLabs/ModSecurity && \
+    git -C /opt/ModSecurity submodule update --init --recursive && \
+    (cd "/opt/ModSecurity" && \
+        ./build.sh && \
+        ./configure && \
+        make && \
+        make install \
+    ) && \
+    rm -fr /opt/ModSecurity \
+        /usr/local/modsecurity/lib/libmodsecurity.a \
+        /usr/local/modsecurity/lib/libmodsecurity.la
+
+# Clone Modsec Nginx Connector, ModSec OWASP Rules, and download/extract nginx and GeoIP databases
+RUN echo 'Cloning Modsec Nginx Connector and ModSec OWASP Rules' && \
+    git clone -b master --depth 1 https://github.com/SpiderLabs/ModSecurity-nginx.git && \
+    git clone -b ${OWASP_BRANCH} --depth 1 https://github.com/coreruleset/coreruleset.git /usr/local/owasp-modsecurity-crs && \
+    wget -O - https://nginx.org/download/nginx-${NGINX_VERSION}.tar.gz | tar -xz
+
+# Install ModSecurity Nginx modules
+# Since Alpine nginx package is not compiled using
+# `--with-compat` flag, we must specify all the same
+# compile options that are used to compile the nginx Alpine
+# packge when we configure the modsecurity nginx connector
+# https://git.alpinelinux.org/aports/tree/main/nginx/APKBUILD?h=3.11-stable
```

Comment:
Oh, good grief. That's a pain. Any concerns about keeping in-sync with potential changes in the base build?

Reply:
I've open an [issue in the Alpine repo](https://gitlab.alpinelinux.org/alpine/aports/-/issues/12375) to request it be added upstream. This "should" only be a concern, if the nginx version in the addon is updated before the `--with-compat` flag is added upstream (assuming it will be added at some point). It really is a 1 line change to add `--with-compat` with the rest of the existing options, so I'm hoping it won't take too much time for the Alpine maintainer for the nginx package to evaluate adding it.

I've seen some people dynamically determine the required compile options by installing nginx from package manager, dumping the compile-time options, `sed` out the stuff that's not needed, and then add those options to the `./configure` command, but that seems like a workaround for something that can/should be fixed upstream.

Reply:
I understand. It's as good as we can hope right now. Luckily, we can control our destiny somewhat with build.json here so an "unprompted" upstream change shouldn't impact us.

I'm curious to see whether that issue gets picked up.

Reply:
I really don't see why it wouldn't. I attached a link to a similar request for the Debian nginx package that has already been accepted and merged so hopefully that will help speed the consideration to see that it has already been implemented in other distros, including the code change required to implement it. The Alpine maintainer for the nginx package seems pretty active and there is usually a couple of commits added about every month or two. I'll open a PR for it if it doesn't get picked up before I get a chance to.

Reply:
looks someone has opened merge request in the Alpine repo to add --with-compat flag. are we good to go ahead and resolve this thread and the other one?

Reply:
Sure thing, although it'll be some time before this change gets pushed down to a home assistant base-image.